src/auth/verify.rs

```diff
@@ -12,79 +14,146 @@ pub(crate) trait Verify {
     async fn verify(&self, req_ctx: &RequestContext) -> Result<(), Error>;
 }
 
-#[allow(clippy::large_enum_variant)]
-// The difference in size is indeed significant here
-// but it's quite unlikely that someone will require to store several hundreds
-// of providers or more to care much
-pub enum AuthVerifier {
+pub enum Verifier {
     Basic(BasicVerifier),
     Jwt(JwtVerifier),
-    Or(Box<AuthVerifier>, Box<AuthVerifier>),
-    And(Box<AuthVerifier>, Box<AuthVerifier>),
 }
 
-impl AuthVerifier {
-    pub fn new(provider: blueprint::Auth) -> Option<Self> {
+pub enum AuthVerifier {
+    Single(Verifier),
+    Any(Vec<AuthVerifier>),
+    All(Vec<AuthVerifier>),
+}
```

Comment:
Ideally, we would want invalid states to be irrepresentable. That's not happening here. One could create `AuthVerifier::Any(vec![])` and `AuthVerifier::All(vec![])` and these are invalid states, requiring additional checks.

Reply:
I think this has well-established behaviour for functions like all and any https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all